src/osd/scrubber/scrub_queue.h

```diff
@@ -0,0 +1,199 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include "include/expected.hpp"
+#include "include/utime_fmt.h"
+#include "osd/osd_types.h"
+#include "osd/osd_types_fmt.h"
+#include "osd/scrubber/scrub_queue_if.h"
+#include "osd/scrubber_common.h"
+
+#include "utime.h"
+
+
+namespace Scrub {
+class ScrubSchedListener;
+class ScrubJob;
+class SchedEntry;
+}  // namespace Scrub
+
+/**
+ * the queue of PGs waiting to be scrubbed.
+ *
+ * the following invariants hold:
+ * - there are at most two objects for each PG (one for each scrub type) in
+ *   the queue.
+ * - if a queue element is removed or white-out, the corresponding object held
+ *   by the PgScrubber will (not necessarily immediately) be marked as
+ *   'not in the queue'.
+ * - 'white-out' queue elements are never reported to the queue users.
```

Comment:
white-out is an implementation detail and does not belong here as a user of this interface should never observe behavior related to an item having been whited-out.